x-pack/plugins/reporting/server/config/schema.ts

```diff
@@ -138,6 +145,7 @@ export const ConfigSchema = schema.object({
   roles: RolesSchema,
   poll: PollSchema,
   export_types: ExportTypeSchema,
+  disableStatefulSettings: SettingsSchema,
```

Comment:
Naming-wise, it often creates headaches when folks see a boolean setting name based on a negative, like `disableX`. I suggest we make the new setting name `enableStatefulSettings`. In serverless, the value will be `false` and in stateful the value will be `true.